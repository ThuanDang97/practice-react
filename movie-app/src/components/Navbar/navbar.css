.dropdown-menu {
  display: flex;
  gap: 50px;
}

li span {
  color: white;
  font-weight: 700;
}

.dropdown-content {
  flex-direction: column;
  display: none;
  z-index: 10;
}

.dropdown-item:hover .dropdown-content {
  max-height: 398px;
  overflow: auto;
  position: absolute;
  font-size: 16px;
  font-stretch: 100%;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  display: flex;
  transform: translateY(0px);
  background-color: aliceblue;
  border-radius: 5px;
  padding: 5px 0;
}
