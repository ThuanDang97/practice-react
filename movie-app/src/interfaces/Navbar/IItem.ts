export interface IItem {
  idItem: number;
  titleItem: string;
  pathItem: string;
}
