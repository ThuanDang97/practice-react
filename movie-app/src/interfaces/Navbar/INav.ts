import { IItem } from './IItem';

export interface INavbar {
  id: number;
  title: string;
  items: IItem[];
  path: string;
}
