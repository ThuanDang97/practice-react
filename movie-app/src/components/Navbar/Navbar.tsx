import { Link } from 'react-router-dom';
import { INavbar } from '../../interfaces/Navbar/INav';
import './navbar.css';
import { NavItem } from './NavItem/NavItem';

type TProps = {
  navItem: INavbar[];
};

const Navbar = (props: TProps): JSX.Element => {
  const { navItem } = props;
  return (
    <>
      <nav>
        <ul className="dropdown-menu">
          {navItem.map((item) => (
            <li key={item.id} className="dropdown-item">
              <Link to={item.path}>
                <span>{item.title}</span>
              </Link>
              {item.items.length > 0 && <NavItem itemNav={item.items} />}
            </li>
          ))}
        </ul>
      </nav>
    </>
  );
};

export { Navbar };
