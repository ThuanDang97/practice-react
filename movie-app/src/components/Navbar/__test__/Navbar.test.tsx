import { render, screen } from '@testing-library/react';
import { Navbar } from '../Navbar';
import { BrowserRouter as Router } from 'react-router-dom';
describe('Test Navbar component', () => {
  const mockItem = [
    {
      id: 1,
      title: 'Nav Item',
      items: [],
      path: '/',
    },
  ];

  const component = (
    <Router>
      <Navbar navItem={mockItem} />
    </Router>
  );

  test('Component Navbar matches DOM Snapshot', () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Navbar should render correctly', () => {
    render(component);

    expect(screen.getByTestId('navbar')).toHaveTextContent('Nav Item');
  });
});
