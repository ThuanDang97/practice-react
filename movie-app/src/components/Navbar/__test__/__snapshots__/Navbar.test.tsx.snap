// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar component Component Navbar matches DOM Snapshot 1`] = `
<div>
  <nav
    data-testid="navbar"
  >
    <ul
      class="dropdown-menu"
    >
      <li
        class="dropdown-item"
      >
        <a
          href="/"
        >
          <span>
            Nav Item
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
